File: src/main/java/org/poglebiarka/app/notifications/NotificationRequest.java
package org.poglebiarka.app.notifications;

import jakarta.validation.constraints.NotBlank;

import java.util.List;

public record NotificationRequest(
        @NotBlank
        String mainTitle,
        @NotBlank
        String mainDescription,
        @NotBlank
        String subTitle,
        @NotBlank
        String subDescription,
        List<Link> links) {

}

File: src/main/kotlin/org/poglebiarka/admin/notifications/NotificationRequest.kt
package org.poglebiarka.admin.notifications

import jakarta.validation.constraints.NotBlank

data class NotificationRequest(
    val mainTitle: @NotBlank String,
    val mainDescription: @NotBlank String,
    val subTitle: @NotBlank String,
    val subDescription: @NotBlank String,
    val links: List<Link>
)

class Link(val title: String, val url: String)

